commons/src/main/java/org/infinispan/commons/util/SmallIntSet.java

```diff
@@ -87,24 +87,17 @@ public SmallIntSet(int initialRange) {
       bitSet = new BitSet(initialRange);
    }
 
-   public SmallIntSet(SmallIntSet other) {
-      bitSet = new BitSet(other.bitSet.size());
-      bitSet.or(other.bitSet);
-   }
-
    public SmallIntSet(Set<Integer> set) {
-      bitSet = new BitSet();
-      set.forEach(bitSet::set);
-   }
-
-   public SmallIntSet(IntSet set) {
       if (set instanceof SmallIntSet) {
          BitSet bitSet = ((SmallIntSet) set).bitSet;
          this.bitSet = new BitSet(bitSet.size());
          this.bitSet.or(bitSet);
-      } else {
+      } else if (set instanceof IntSet) {
          this.bitSet = new BitSet();
-         set.forEach((IntConsumer) bitSet::set);
+         ((IntSet) set).iterator().forEachRemaining((IntConsumer) bitSet::set);
```

Comment:
`forEach`

Reply:
This is what happens when I add this change in parallel :)